op-node/p2p/pings_test.go

```diff
@@ -0,0 +1,79 @@
+package p2p
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slog"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+)
+
+func TestPingService(t *testing.T) {
+	peers := []peer.ID{"a", "b", "c"}
+	log, captLog := testlog.CaptureLogger(t, slog.LevelDebug)
+
+	testCase := 0
```

Comment:
nit: consider calling this "count" or something, `testCase` makes it sound like we're running multiple separate tests here, but really this is just a call counter to control behavior.